File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.py
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def calc(node, left):
            nonlocal result
            if not node.left and not node.right and left:
                result += node.val
            if node.left:
                calc(node.left, True)
            if node.right:
                calc(node.right, False)

        result = 0
        calc(root, False)
        return result

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.cpp
class Solution {
private:
    int result;

    void calc(TreeNode* node, bool left) {
        if (node->left == NULL && node->right == NULL && left) {
            result += node->val;
        }
        if (node->left != NULL) {
            calc(node->left, true);
        }
        if (node->right != NULL) {
            calc(node->right, false);
        }
    }

public:
    int sumOfLeftLeaves(TreeNode* root) {
        result = 0;
        calc(root, false);
        return result;
    }
};